src/coord/cartesian/Cartesian2D.ts

```diff
@@ -105,6 +105,17 @@ class Cartesian2D extends Cartesian<Axis2D> implements CoordinateSystem {
             && this.getAxis('y').containData(data[1]);
     }
 
+    containZone(data1: ScaleDataValue[], data2: ScaleDataValue[]): boolean {
+        const zoneDiag1 = this.dataToPoint(data1);
+        const zoneDiag2 = this.dataToPoint(data2);
+        const area = this.getArea();
+        const zoneX = Math.min(zoneDiag1[0], zoneDiag2[0]);
```

Comment:
We can use the `zoneDiag1` and `zoneDiag2` directly when creating `BoundingRect`. It will handle min, max logic in the constructor:

https://github.com/ecomfe/zrender/blob/master/src/core/BoundingRect.ts#L27

Reply:
Thank you for pointing this out! Zrender does have some handful functions I ignored.